packages/main/test/specs/DatePicker.spec.js

```diff
@@ -1345,4 +1345,28 @@ describe("Date Picker Tests", () => {
 
 		assert.strictEqual(await input.getProperty("valueState"), valueState, "value state is not changed");
 	});
+
+	it("should open date picker in daypicker", async () => {
+		datepicker.id = "#dpCalendarModeMonths";
+
+		const calendar = await datepicker.getCalendar();
+		const datepickerRoot = await datepicker.getRoot();
+		await datepicker.openPicker();
+
+		let currentPicker = await calendar.getProperty("_currentPicker");
+		assert.equal(currentPicker, "month", "calendar is opened on months");
+		
+		await datepickerRoot.setAttribute("format-pattern", "yyyy, dd/MM");
+		await datepicker.openPicker();
+		currentPicker = await calendar.getProperty("_currentPicker");
+
+		assert.equal(currentPicker, "day", "calendar is opened on months");
```

Comment:
It looks good to me. Tested in different cases as setting the format-pattern programatically, manually and it seems to always open the correct picker.

Minor one tho, 

> calendar is opened on months -> days